### DISCUSSION TITLE
Upload an image using ui.upload to MySQL Database as BLOB

### DISCUSSION THREAD

**Discussion by BigPesh** [zauberzeug/nicegui#1278] [Q&A] [answered]
### Question

Hey All,

I'm currently using ui.upload to let the user select or take an image, I then want to save the image to a MySQL database, I normally do this by converting the image to base64 and save it in the database as a BLOB. Where does the image file get saved when you click the upload button?

Thanks

**Comment by falkoschindler** [comment, marked as answer]
Hi @BigPesh,

The image is not saved unless you do so. From within your event handler you can access the base64 data like this:

```py
def handle_upload(e: events.UploadEventArguments):
    base64_data = base64.b64encode(e.content.read())
    ...

ui.upload(on_upload=handle_upload)
```

**Reply by BigPesh** [reply]
Works perfect thank you very much for this!
